### PR TITLE
feat: Require NEAR regular RPC URL additionally in configs

### PR DESCRIPTION
**Pull request by kobayurii**
This pull request is a comprehensive enhancement that resolves the existing issue pertaining to NEAR_RPC_URL configuration and optimizes Remote Procedure Call (RPC) handling within the project.

1. Added NEAR_RPC_URL as the Regular RPC endpoint.
2. Added ARCHIVAL_NEAR_RPC_URL as an optional setting for Archival RPC endpoints.
3. Added logic to utilize Archival RPC selectively for specific use cases, like shadow_data_consistency.

This pull request aims to improve RPC configurations and usage, enhancing the project's efficiency and performance while providing more granular control over RPC endpoints for different functionalities.

### CODE REVIEW

**Comment by kobayurii** [comment]
https://github.com/near/read-rpc/issues/136

File: rpc-server/src/config.rs [hunk 1]
```diff
@@ -8,6 +8,15 @@ pub struct Opts {
     #[clap(long, env = "NEAR_RPC_URL")]
     pub rpc_url: http::Uri,
 
+    // near network archive rpc url
+    #[clap(long, env = "ARCHIVE_NEAR_RPC_URL")]
```
Comment (khorolets) [inline rpc-server/src/config.rs:12]: `ARCHIVAL_NEAR_RPC_URL` please

File: rpc-server/src/config.rs [hunk 2]
```diff
@@ -8,6 +8,15 @@ pub struct Opts {
...
+    pub archive_rpc_url: Option<http::Uri>,
```
Comment (khorolets) [inline rpc-server/src/config.rs:13]:
```suggestion
    pub archival_rpc_url: Option<http::Uri>,
```

File: rpc-server/src/modules/queries/methods.rs [hunk 3]
```diff
@@ -76,10 +75,15 @@ async fn query_call(
         near_primitives::views::QueryRequest::ViewState {
             account_id,
             prefix,
-            include_proof: _,
+            include_proof,
         } => {
             crate::metrics::QUERY_VIEW_STATE_REQUESTS_TOTAL.inc();
-            view_state(&data, block, &account_id, prefix.as_ref()).await
+            if include_proof {
+                // We can calculate the proof for state only on archive nodes.
```
Comment (khorolets) [inline rpc-server/src/modules/queries/methods.rs:82]: This comment is not true. We can calculate proofs on regular nodes as well. What we cannot do is we cannot calculate it on ReadRPC.

Put a TODO here, please. Ideally, we would like to differentiate regular-data requests from the historical ones and proxy such requests to a corresponding NEAR RPC URL, but for now, let's proxy all such requests to the Archival node.

File: rpc-server/src/utils.rs [hunk 4]
```diff
@@ -7,21 +7,75 @@ use sysinfo::{System, SystemExt};
...
-    client: &near_jsonrpc_client::JsonRpcClient,
-    params: M,
-) -> near_jsonrpc_client::MethodCallResult<M::Response, M::Error>
-where
-    M: near_jsonrpc_client::methods::RpcMethod + std::fmt::Debug,
-{
-    tracing::debug!("PROXY call. {:?}", params);
-    client.call(params).await
+#[derive(Clone)]
+pub struct JsonRpcClient {
```
Comment (khorolets) [inline rpc-server/src/utils.rs:11]: Doc-string explaining this would be nice for future us.

**Review by khorolets — CHANGES_REQUESTED** [review]
Looks good! Please, improve the naming, it's not "archive" but "archival"

File: rpc-server/src/modules/queries/methods.rs [hunk 5]
```diff
@@ -76,10 +75,18 @@ async fn query_call(
...
             prefix,
-            include_proof: _,
+            include_proof,
         } => {
             crate::metrics::QUERY_VIEW_STATE_REQUESTS_TOTAL.inc();
-            view_state(&data, block, &account_id, prefix.as_ref()).await
+            if include_proof {
+                // TODO: We can calculate the proof for state only on regular or archival nodes.
+                // After indexing the epochs and validators,
+                // we will be able to separate proxies queries to regular and archive nodes.
```
Comment (khorolets) [inline rpc-server/src/modules/queries/methods.rs:84]:
```suggestion
                // we will be able to separate proxies queries to regular and archival nodes.
```

File: rpc-server/src/modules/queries/methods.rs [hunk 6]
```diff
@@ -76,10 +75,18 @@ async fn query_call(
...
+                // For now we will proxy all requests with `include_proof` to archive nodes.
```
Comment (khorolets) [inline rpc-server/src/modules/queries/methods.rs:85]:
```suggestion
                // For now we will proxy all requests with `include_proof` to archival nodes.
```

File: rpc-server/src/utils.rs [hunk 7]
```diff
@@ -7,21 +7,84 @@ use sysinfo::{System, SystemExt};
...
-    M: near_jsonrpc_client::methods::RpcMethod + std::fmt::Debug,
-{
-    tracing::debug!("PROXY call. {:?}", params);
-    client.call(params).await
+/// JsonRpcClient represents a client capable of interacting with NEAR JSON-RPC endpoints,
+/// The client is capable of handling requests to both regular and archival nodes.
+#[derive(Clone)]
+pub struct JsonRpcClient {
+    regular_client: near_jsonrpc_client::JsonRpcClient,
+    archive_client: near_jsonrpc_client::JsonRpcClient,
```
Comment (khorolets) [inline rpc-server/src/utils.rs:15]:
```suggestion
    archival_client: near_jsonrpc_client::JsonRpcClient,
```

File: rpc-server/src/utils.rs [hunk 8]
```diff
@@ -7,21 +7,84 @@ use sysinfo::{System, SystemExt};
...
+    ) -> near_jsonrpc_client::MethodCallResult<M::Response, M::Error>
+    where
+        M: near_jsonrpc_client::methods::RpcMethod + std::fmt::Debug,
+    {
+        tracing::debug!("PROXY call. {:?}", params);
+        self.rpc_call(params, false).await
+    }
+
+    /// Performs an RPC call to the archival endpoint.
+    pub async fn archive_call<M>(
```
Comment (khorolets) [inline rpc-server/src/utils.rs:75]:
```suggestion
    pub async fn archival_call<M>(
```

**Review by khorolets — CHANGES_REQUESTED** [review]

**Review by khorolets — DISMISSED** [review]

**Review by khorolets — APPROVED** [review]

**Pull request merged**